src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs

```diff
@@ -242,16 +284,107 @@ private void WriteClientMethods(CodeWriter writer, ResourceOperation resourceOpe
                     if (container == null)
                         return;
                     writer.Line();
-                    writer.WriteXmlDocumentationSummary($"Gets a list of {container.ResourceName} in the {resourceOperation.ResourceName}.");
-                    writer.WriteXmlDocumentationReturns($"An object representing collection of {StringExtensions.Pluralization(container.ResourceName)} and their operations over a {resourceOperation.ResourceName}.");
-                    using (writer.Scope($"public {container.Type} Get{StringExtensions.Pluralization(container.ResourceName)}()"))
+                    writer.WriteXmlDocumentationSummary($"Gets a list of {container.ResourceName.ToPlural()} in the {resourceOperation.ResourceName}.");
+                    writer.WriteXmlDocumentationReturns($"An object representing collection of {container.ResourceName.ToPlural()} and their operations over a {resourceOperation.ResourceName}.");
+                    using (writer.Scope($"public {container.Type} Get{container.ResourceName.ToPlural()}()"))
                     {
                         writer.Line($"return new {container.Type}(this);");
                     }
                 }
             }
         }
 
+        private void WriteChildPagingMethod(CodeWriter writer, OperationGroup operationGroup, bool async, PagingMethod pagingMethod)
+        {
+            writer.Line();
+            Parameter[] nonPathParameters = GetNonPathParameters(pagingMethod.Method);
+
+            writer.WriteXmlDocumentationSummary(pagingMethod.Method.Description);
+            foreach (var param in nonPathParameters)
+            {
+                writer.WriteXmlDocumentationParameter(param);
+            }
+            writer.WriteXmlDocumentationParameter("cancellationToken", "The cancellation token to use.");
+
+            CSharpType itemType = pagingMethod.PagingResponse.ItemType;
+            string returnText = $"{(async ? "An async" : "A")} collection of <see cref=\"{itemType.Name}\" /> that may take multiple service requests to iterate over.";
+            writer.WriteXmlDocumentation("returns", returnText);
+
+            var returnType = itemType.WrapPageable(async);
+
+            var methodName = CreateMethodName(pagingMethod.Name, async);
+            writer.Append($"public {returnType} {methodName}(");
+            foreach (var param in nonPathParameters)
+            {
+                writer.WriteParameter(param);
+            }
+            writer.Line($"{typeof(CancellationToken)} cancellationToken = default)");
+
+            using (writer.Scope())
+            {
+                WritePagingMethodImplementation(writer, operationGroup, pagingMethod, async, itemType);
+            }
+        }
+
+        private void WritePagingMethodImplementation(CodeWriter writer, OperationGroup operationGroup, PagingMethod pagingMethod, bool async, CSharpType itemType)
```

Comment:
Investigate if it is possible to reuse the `WritePagingMethod` in `MgmtClientBaseWriter`